### PR TITLE
Scripts to build and merge Seek DBs

### PR DESCRIPTION
**Pull request by gdoubleyew**
This PR builds out Python scripts to help with creating and merging Seek DBs.

- seekCreateDB.py - create a new Seek DB from PCL files.
- seekIncrementalMerge.py - merge in new PCL datasets files into an existing smallDB. Where there is a small and largeDB and new datasets are merged into the small one.
- seekUtils.py - parallelized functions for building a Seek database. These include builds of the main DB files as well as metadata files.
- runParallelJobs.py - a helper script for running a queue of tasks in parallel.
- Add test scripts

### CODE REVIEW

File: scripts/seek/mapDsetToPlatform.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    'Mouse430_2': 'GPL1261',
+    'Yeast_2': 'GPL2529',
+    'U133_X3P': 'GPL1352',
+    'E_coli_2': 'GPL3154',
+    'wheat': 'GPL3802',
+    'Celegans': 'GPL200',
+    'Xenopus_laevis': 'GPL1318',
+    'Mapping10K_Xba131': 'GPL1266',
+    'Mapping10K_Xba142': 'GPL2641',
+    'MG_U74C': 'GPL83',
```
Comment (atadych) [inline scripts/seek/mapDsetToPlatform.py:69]: Duplicate key: 'MG_U74C'

File: scripts/seek/runParallelJobs.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            args = job.get('args', ())
+            kwargs = job.get('kwargs', {})
+            func = job.get('func', None)
+            if func is None:
+                raise ValueError('Function not defined')
+            print(f'Thread {tid}: func: {func.__name__}, args: {args}, kwargs: {kwargs}')
+            func(*args, **kwargs)
+    except queue.Empty:
+        pass
+    return
```
Comment (atadych) [inline scripts/seek/runParallelJobs.py:24]: Not sure about this return. But maybe later you would like to add a status code?

**Review by atadych — COMMENTED** [review]
I ran all the tests in seek/tests and all were successful except TestSeekCreateDB.test_createDB

**Comment by atadych** [comment]
Regarding `test_createDB`, I am getting segmentation when `SeekMiner` command is run

`/Users/alicja/dev/workspace/sleipnir/gw_scripts/Debug/SeekMiner -x /tmp/testSeekDb/mockDb/dset_plat_map.txt -i /tmp/testSeekDb/mockDb/gene_map.txt -d /tmp/testSeekDb/mockDb/db -p /tmp/testSeekDb/mockDb/prep -P /tmp/testSeekDb/mockDb/plat -Q /tmp/testSeekDb/mockDb/quant2 -u /tmp/testSeekDb/mockDb/sinfo -n 100 -b 200  -V CV -I LOI -z z_score -m -M -O -q /tmp/testSeekDb/mockDb/query.txt -o /tmp/testSeekDb/mockDb/results -Y -T 2 -t 1`

Output:

```
Start reading average and presence files
1616766773N
Done reading average and presence files
1616766773N
Initializing gene map
1616766773N
Done initializing gene map
1616766773N
Initializing query map
Done initializing query map
Reading 2 query genes' correlations
1616766773N
Finished reading query genes' correlations
1616766773N
Done search
1616766773N
Segmentation fault: 11
```

File: scripts/seek/runParallelJobs.py (hunk 2 above)
Comment (gdoubleyew) [reply scripts/seek/runParallelJobs.py:24]: Yes, that's a good point. I have a return queue for jobThread, maybe I should add one here also.

File: scripts/seek/seekUtils.py [hunk 3]
```diff
@@ -0,0 +1,461 @@
...
+        if len(cols) > 3:
+            print(f"Error line({count}): contains too many columns!\n");
+            break
+        else:
+            dset_list.append(tuple(cols))
+    fp.close()
+    return dset_list
+
+
+def write_dataset_list(dset_list, dsetFile):
```
[the remark this answers is not in the document]
Comment (gdoubleyew) [reply scripts/seek/seekUtils.py:120]: Yes, good point, I copied those over from the script in the download for the breast cancer example. I'll rename them.

File: scripts/seek/mapDsetToPlatform.py (hunk 1 above)
Comment (gdoubleyew) [reply scripts/seek/mapDsetToPlatform.py:69]: Good catch, thanks! This one is not only duplicate, but also wrong. It duplicates 'MG_U74C': 'GPL34', but was probably intended as 'MG_U74Cv2': 'GPL83',

File: scripts/seek/tests/test_seekCreateDB.py [hunk 4]
```diff
@@ -22,7 +22,7 @@ def setup_class(cls):
             cls.temp_dir = tempfile.TemporaryDirectory()
             tmpDirName = cls.temp_dir.name
         else:
-            tmpDirName = '/tmp/testSeekUtils'
+            tmpDirName = '/tmp/testSeekCreateDb'
```
Comment (atadych) [inline scripts/seek/tests/test_seekCreateDB.py:25]: Funny, because locally I changed my temp dir to this path as well to distinguish it from other tests.

**Review by atadych — APPROVED** [review]
Looks good to me

**Pull request merged**